addons/nightvision/XEH_postInit.sqf

```diff
@@ -27,12 +27,34 @@ GVAR(isUsingMagnification) = false;
     TRACE_4("settingsInitialized",GVAR(disableNVGsWithSights),GVAR(fogScaling),GVAR(noiseScaling),GVAR(effectScaling));
 
     ["visionMode", LINKFUNC(onVisionModeChanged), false] call CBA_fnc_addPlayerEventHandler;
-    ["loadout", LINKFUNC(onLoadoutChanged), true] call CBA_fnc_addPlayerEventHandler;
     ["cameraView", LINKFUNC(onCameraViewChanged), true] call CBA_fnc_addPlayerEventHandler;
     ["vehicle", LINKFUNC(refreshGoggleType), false] call CBA_fnc_addPlayerEventHandler;
     ["turret", LINKFUNC(refreshGoggleType), true] call CBA_fnc_addPlayerEventHandler;
     ["ACE_controlledUAV", LINKFUNC(refreshGoggleType)] call CBA_fnc_addEventHandler;
 
+    ["unit", {
+        params ["_newPlayer", "_oldPlayer"];
+
+        private _ehID = _oldPlayer getVariable QGVAR(ehID);
+
+        if (!isNil "_ehID") then {
+            _oldPlayer removeEventHandler ["SlotItemChanged", _ehID];
+
+            _oldPlayer setVariable [QGVAR(ehID), nil];
+        };
+
+        _ehID = _newPlayer getVariable QGVAR(ehID);
+
+        if (isNil "_ehID") then {
+            // Call manually to update existing value
+            [] call FUNC(refreshGoggleType);
+
+            _ehID = _newPlayer addEventHandler ["SlotItemChanged", LINKFUNC(onSlotItemChanged)];
```

Comment:
I could update a lot of this to use `CBA_fnc_addBISPlayerEventHandler`
but it doesn't really matter

Reply:
Imo a good idea, for consistency's sake.